schema.production.kf

```diff
@@ -330,6 +299,75 @@ action remove_credential($id) public {
     );
 }
 
+action share_credential(
+        $id,
+        $original_credential_id,
+        $credential_type,
+        $credential_level,
+        $credential_status,
+        $content,
+        $encryption_public_key) public {
+    $converted = idos_near.implicit_address_to_public_key(@caller);
+    SELECT CASE
+    WHEN NOT EXISTS (
+        SELECT 1 from credentials
+            WHERE id = $original_credential_id
+            AND human_id=(SELECT DISTINCT human_id FROM wallets WHERE (wallet_type = 'EVM' AND address=@caller COLLATE NOCASE)
+                OR (wallet_type = 'NEAR' AND public_key = $converted))
+    ) THEN ERROR('The original credential does not belong to the caller') END;
+
+    add_credential($id, $credential_type, $credential_level, $credential_status, $content, $encryption_public_key);
+    INSERT INTO shared_credentials (original_id, duplicate_id) VALUES ($original_credential_id, $id);
+}
+
+action share_credential_through_dag (
+        $id,
+        $human_id,
+        $original_credential_id,
+        $credential_type,
+        $credential_level,
+        $credential_status,
+        $content,
+        $encryption_public_key,
+        $dag_owner,
+        $dag_grantee,
+        $dag_locked_until,
+        $dag_signature) public {
+    SELECT CASE
+    WHEN NOT EXISTS (
+        SELECT 1 from credentials
+            WHERE id = $original_credential_id
+            AND human_id = $human_id
+    ) THEN ERROR('The original credential does not belong to the human') END;
+
+    $owner_verified = idos_eth.verify_owner($dag_owner, $dag_grantee, $id, $dag_locked_until, $dag_signature);
+    SELECT CASE
+        WHEN $owner_verified != 1 THEN ERROR('the signature is invalid')
+    END;
+
+    SELECT CASE
+    WHEN NOT EXISTS (
+        SELECT 1 from humans
+            INNER JOIN wallets ON humans.id = wallets.human_id
+            WHERE wallet_type = 'EVM'
+            AND address=$dag_owner COLLATE NOCASE
+            AND human_id = $human_id
```

Comment:
There's a clear lack of NEAR on this. I'm not sure how much this breaks current expectations.

Let's make sure that Product is ok with this trade-off, and let's add a task to support NEAR.

Reply:
This is an acceptable trade-off for now. If we hear a lot of concerns from other issuers about not being able to call this on the user's behalf, then we'll also implement NEAR.